src/components/settings/PersonalizedPushSettings.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useContext, useEffect, useMemo, useState } from 'react';
+import { useQuery } from 'react-apollo';
+import { FlatList, StyleSheet } from 'react-native';
+
+import { SettingsContext } from '../../SettingsProvider';
+import { normalize, texts } from '../../config';
+import { getInAppPermission } from '../../pushNotifications';
+import { getQuery, QUERY_TYPES } from '../../queries';
+import { onActivatePushNotifications, onDeactivatePushNotifications } from '../../screens';
+import { LoadingSpinner } from '../LoadingSpinner';
+import { SafeAreaViewFlex } from '../SafeAreaViewFlex';
+import { SettingsToggle } from '../SettingsToggle';
+
+type Category = { id: string; iconName?: string | null; name: string };
+
+export const PersonalizedPushSettings = () => {
+  const { globalSettings } = useContext(SettingsContext);
+  const { settings = {} } = globalSettings;
+  const { personalizedPush = {}, settingsScreenListItemTitles = {} } = settings;
+  const queryVariables = personalizedPush.variables || {
+    tagList: ['news_item']
+  };
+
+  const [permission, setPermission] = useState<boolean>(false);
+
+  const { data, loading } = useQuery(getQuery(QUERY_TYPES.CATEGORIES_FILTER), {
+    variables: queryVariables,
+    skip: !personalizedPush?.categories
+  });
+
+  useEffect(() => {
+    (async () => {
+      const inAppPermission = await getInAppPermission();
+      setPermission(!!inAppPermission);
+    })();
+  }, []);
+
+  const handleActivate = (revert: () => void) => {
+    setPermission(true);
+    onActivatePushNotifications(() => {
+      setPermission(false);
+      revert?.();
+    });
+  };
+
+  const handleDeactivate = (revert: () => void) => {
+    setPermission(false);
+    onDeactivatePushNotifications(() => {
+      setPermission(true);
+      revert?.();
+    });
+  };
+
+  const listItems = useMemo(() => {
+    const categories: Category[] = data?.categories ?? [];
+
+    const initialListItem = {
+      onActivate: handleActivate,
+      onDeactivate: handleDeactivate,
```

Comment:
why calling them `handle...` and not `on...` like `onActivate: onActivatePushNotifications` and `onDeactivate: onDeactivatePushNotifications`? i will update it